datafusion/datasource/src/source.rs

```diff
@@ -372,6 +378,20 @@ impl DataSourceExec {
         self
     }
 
+    /// Add filters' equivalence info
+    fn add_filter_equivalence_info(
+        mut self,
+        filter: Arc<dyn PhysicalExpr>,
+    ) -> Result<Self> {
+        let (equal_pairs, _) = collect_columns_from_predicate(&filter);
+        for (lhs, rhs) in equal_pairs {
+            self.cache
+                .eq_properties
+                .add_equal_conditions(Arc::clone(lhs), Arc::clone(rhs))?
+        }
+        Ok(self)
+    }
```

Comment:
This part I'm not familiar with and would appreciate if someone else took a look at and confirm is correct.

Reply:
Basically, it's enough, lgtm